### PR TITLE
fix: subset & export to encord button.

### PR DESCRIPTION
**Pull request by james-encord**
Subset: no longer crashes, generated project seems to work.

Export to encord: fixed via ignoring hash mappings. Double checking db hash remapping behaviour at the momement.

Found bug with exporting a subset project to encord failing, investigating currently.

### CODE REVIEW

**Comment by james-encord** [comment]
Status: almost fully working except that the db migration step errors with:

ConnectorError(ConnectorError { user_facing_error: None, kind: QueryError(SqliteFailure(Error { code: ReadOnly, extended_code: 1032 }, Some("attempt to write a readonly database"))), transient: false })

Currently investigating

**Comment by james-encord** [comment]
Works - digging into bizarre behaviour of prisma regarding caching connections long term.

File: src/encord_active/server/main.py [hunk 1]
```diff
@@ -11,7 +11,7 @@
 
 app.include_router(project.router)
 
-origins = ["http://localhost:5173", "http://localhost:8501", get_settings().ALLOWED_ORIGIN]
+origins = ["http://localhost:5173", "http://localhost:8501", "http://localhost:8000", get_settings().ALLOWED_ORIGIN]
```
Comment (Encord-davids) [inline src/encord_active/server/main.py:14]: [curious]🤔

Not sure why we need this one

File: src/encord_active/lib/encord/project_sync.py [hunk 2]
```diff
@@ -226,7 +228,45 @@ def copy_filtered_data(
...
+                                if data_unit.data_uri is not None
+                                else None,
+                                "width": data_unit.width,
+                                "height": data_unit.height,
+                                "fps": data_unit.fps,
+                            }
+                            for data_unit in (label_row_db.data_units or [])
+                        ]
+                    },
+                    "label_row_json": json.dumps(label_row_mapping[label_row_db.label_hash or ""], indent=2),
```
Comment (Encord-davids) [inline src/encord_active/lib/encord/project_sync.py:261]: [nitpick] 💅

We probably shouldn't add indentation to keep it compact 

```suggestion
                    "label_row_json": json.dumps(label_row_mapping[label_row_db.label_hash or ""]),
```

File: src/encord_active/lib/project/project_file_structure.py [hunk 3]
```diff
@@ -59,27 +59,42 @@ def get_encord_project_cached(ssh_key_path: str, project_hash: str) -> encord.Pr
...
-                label_row_json = entry.label_row_json
-            if label_row_json is None:
-                raise ValueError("label_row_json does not exist")
-            return json.loads(label_row_json)
+            res = conn.labelrow.find_unique(where={"label_hash": self._label_hash})
+            if res is None:
+                raise ValueError(f"label_row missing in prisma db(label_hash={self._label_hash})")
+            return res
+
+    def get_label_row_json(self, cache_db: Optional[prisma.Prisma] = None) -> Dict[str, Any]:
```
Comment (Encord-davids) [inline src/encord_active/lib/project/project_file_structure.py:90]: [nitpick] 💅

We could probably make this a `@property`

File: src/encord_active/lib/project/project_file_structure.py [hunk 4]
```diff
@@ -59,27 +59,42 @@ def get_encord_project_cached(ssh_key_path: str, project_hash: str) -> encord.Pr
...
+        self._label_row = label_row
 
     def __hash__(self) -> int:
         return hash(self._label_hash)
 
     def label_row_file_deprecated_for_migration(self) -> Path:
         return self._project.project_dir / "data" / self._label_hash / "label_row.json"
 
-    def get_label_row_json(self, cache_db: Optional[prisma.Prisma] = None) -> Dict[str, Any]:
+    def get_label_row_from_db(self, cache_db: Optional[prisma.Prisma] = None) -> "prisma.models.LabelRow":
```
Comment (Encord-davids) [inline src/encord_active/lib/project/project_file_structure.py:81]: [nitpick] 💅

We could probably make this a `@property`

File: src/encord_active/server/main.py (hunk 1 above)
Comment (james-encord) [reply src/encord_active/server/main.py:14]: i was having cors issues on my local test machine. I can remove from the pr.

File: src/encord_active/lib/project/project_file_structure.py (hunk 3 above)
Comment (james-encord) [reply src/encord_active/lib/project/project_file_structure.py:90]: it is a property, the function is extracted so a cache_db argument can be passed when needed to optimize database connection usage.

File: src/encord_active/lib/project/project_file_structure.py (hunk 4 above)
Comment (james-encord) [reply src/encord_active/lib/project/project_file_structure.py:81]: I'll add a property variant for when cache_db is not needed.

**Review by Encord-davids — APPROVED** [review]

**Pull request merged**